physics/trajectory.hpp

```diff
@@ -0,0 +1,58 @@
+﻿
+#pragma once
+
+#include "base/not_null.hpp"
+#include "geometry/named_quantities.hpp"
+#include "physics/degrees_of_freedom.hpp"
+
+namespace principia {
+namespace physics {
+namespace internal_trajectory {
+
+using base::not_null;
+using geometry::Instant;
+using geometry::Position;
+using geometry::Velocity;
+
+template<typename Frame>
+class Trajectory {
+ public:
+  virtual ~Trajectory() = default;
+
+  // Derived classes of |Trajectory| should also derive |Hint| as appropriate.
+  class Hint {
+   public:
+    virtual ~Hint() = default;
+   protected:
+    Hint() = default;
+  };
+
+  // The time range for which the trajectory can be evaluated.  Note that it is
+  // not required that t_min ≤ t_max: for an empty trajectory, t_min = +∞, and
+  // t_max = -∞.
+  virtual Instant t_min() const = 0;
+  virtual Instant t_max() const = 0;
+
+  // Gets a |Hint| of the appropriate derived type.
+  virtual not_null<std::unique_ptr<Hint>> GetHint() const = 0;
```

Comment:
Call this `NewHint`, it makes it clearer that this is a factory.  Otherwise one is left wondering where it "gets" it from.